src/haz3lcore/dynamics/elaborator.re

```diff
@@ -164,6 +164,68 @@ let rec dhexp_of_uexp = (m: Statics.map, uexp: Term.UExp.t): option(DHExp.t) =>
             tl |> List.fold_left((acc, d) => DHExp.Sequence(d, acc), hd)
           };*/
         Some(EmptyHole(id, 0))
+      | Deferral(true, _) => Some(EmptyHole(id, 0))
+      | Deferral(false, _) => Some(InvalidText(0, 0, "_")) // Unbounded deferral
+      | DeferredAp(fn, arg) =>
```

Comment:
feel like this implementation is a bit awkward... see other comments in maketerm and statics. a lot of syntactic complexity here comes from negotiating the singleton versus tuple case... it might be worth trying to either regularize representation in maketerm (see comment there), or else maybe push the singleton/tuple to the top of the decision tree; like just implement the case of 
`| DeferredAp(fn, arg) when !is_tuple(arg) => ...`
see if there are any obvious helpers, and then hopefully the 
`| DeferredAp(fn,Tuple(es)) => ...`
case just uses the same helpers but maps over things